test/functional/test_framework/key.py

```diff
@@ -444,3 +519,68 @@ def tweak_add(self, tweak):
         ret = ECKey()
         ret.set(tweaked.to_bytes(32, 'big'), self.compressed)
         return ret
+
+def generate_schnorr_nonce():
+    """Generate a random valid Schnorr nonce."""
+    kp = random.randrange(1, SECP256K1_ORDER)
+    assert(kp != 0)
+    R = SECP256K1.affine(SECP256K1.mul([(SECP256K1_G, kp)]))
+    k = kp if jacobi_symbol(R[1], SECP256K1_FIELD_SIZE) == 1 else SECP256K1_ORDER - kp
+    k_key = ECKey()
+    k_key.set(k.to_bytes(32, 'big'), True)
+    return k_key
+
+def aggregate_schnorr_nonces(Rv):
+    """Construct aggregated musig nonce from individually generated nonces."""
+    for idx, R in enumerate(Rv):
+        R_agg = R.p if idx == 0 else SECP256K1.add(R_agg, R.p)
+    R_agg_affine = SECP256K1.affine(R_agg)
+    if jacobi_symbol(R_agg_affine[1], SECP256K1_FIELD_SIZE) != 1:
+        R_agg = SECP256K1.mul([(R_agg, SECP256K1_ORDER - 1)])
+        negated = True
+    else:
+        negated = False
+    ret = ECPubKey()
+    ret.p = R_agg
+    ret.valid = True
+    ret.compressed = True
+    return ret, negated
+
+def generate_musig_key(PubkeyList):
```

Comment:
nit: convention is to use `snake_case` for variables, functions and arguments, and `CamelCase` for class names.